cEP0002.md

```diff
@@ -0,0 +1,272 @@
+# Next Gen Bear Design
+
+| Metadata     |                                         |
+| ------------ |-----------------------------------------|
+| cEP          | 2                                       |
+| Version      | 0.1                                     |
+| Title        | Next gen bear design                    |
+| Authors      | Udayan Tandon <udayan12167@iiitd.ac.in> |
+| Status       | Proposed                                |
+| Type         | Process                                 |
+
+# Abstract
+
+This document describes how the next generation bears will be designed and
+implemented.
+
+This cEP describes a new implementation of bears in coala which should make
+ them simpler and make their design more intuitive and better.
+
+# Introduction
+
+Currently the bear design is clunky and bad. The current design makes global
+bears dependant on local bears through it's queuing mechanism. It uses
+communication signals over a queue to indicate that global bears are done with
+their execution and local bears can begin. The queueing mechanism doesn't work
+very well as it sometimes has synchronization problems. Also if local bears
+fail to run it stalls the global bears. Also we have a permanent logger thread.
+An extra logger thread consumes time and resources of the system which we can
+avoid.
+
+The new proposal aims to eliminate the use of the queuing control signals.
+Utilising a multiprocessing pool to generate the results from the bears. No
+longer will the local bears be dependent on global bears to finish. The logging
+will also be handled by the main thread itself instead of utilising an extra
+thread.
+
+# Proposed Change
+
+Here is the detailed implementation stepwise:
+
+1. coala will have a core multiprocessing pool.
```

Comment:
Yes
On 8 Jul 2016 7:52 p.m., "Udayan Tandon" <notifications@github.com> wrote:

> In cEP0002.md
> <https://github.com/coala-analyzer/cEPs/pull/7#discussion_r70113618>:
>
> > +very well as it sometimes has synchronization problems. Also if local bears
> > +fail to run it stalls the global bears. Also we have a permanent logger thread.
> > +An extra logger thread consumes time and resources of the system which we can
> > +avoid.
> > +
> > +The new proposal aims to eliminate the use of the queuing control signals.
> > +Utilising a multiprocessing pool to generate the results from the bears. No
> > +longer will the local bears be dependent on global bears to finish. The logging
> > +will also be handled by the main thread itself instead of utilising an extra
> > +thread.
> > +
> > +# Proposed Change
> > +
> > +Here is the detailed implementation stepwise:
> > +
> > +1. coala will have a core multiprocessing pool.
>
> Are you talking about the ProcessPoolExecutor in concurrent.Futures?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/cEPs/pull/7/files/8a6ccaf07fc7d713b5a94c66ba17ba2dac6eee5b#r70113618>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFc6KJXuCchdGROC2-U9wdvkJ8LuaaV4ks5qTo5vgaJpZM4Iuahj>
> .
>